### DISCUSSION TITLE
fpv.py Not connecting with WiFi

### DISCUSSION THREAD

**Discussion by matze946** [bitcraze/discussions#626] [Q&A]
Hello, I try to run the fpv.py script to connect to my crazyflie over WiFi and using CPX and CRTP for communication but I get the following error:
`python3 fpv.py
INFO:cflib.crazyflie:Callback->Connection initialized[radio://0/80/2M/E7E7E7E7E7]
INFO:cflib.crazyflie:We are connected[radio://0/80/2M/E7E7E7E7E7], request connection setup
INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
Not connecting with WiFi
INFO:cflib.crazyflie:Closing link
INFO:cflib.crazyflie:Callback->Connected to [radio://0/80/2M/E7E7E7E7E7]
INFO:cflib.crazyflie:Callback->Disconnected from [radio://0/80/2M/E7E7E7E7E7]
Disconnected`

Any ideas? Thanks in advance

**Comment by hmllr** [comment]
Hi :)
Are you connected to the same WiFi as your Crazyflie? Which settings do you use in the config for the WiFi setup at startup (No initial setup, act as AP, connect to a network)?
Best,
Hanna

**Reply by matze946** [reply]
Hi :)
Thanks for your reply. Yes I'm connected to the same WiFi as my Crazyflie. The WiFi is setup to act as an AP. 
Best, Matthias

**Reply by hmllr** [reply]
Hi Matthias,
We actually tested this today - and the same happened to us. It is due to the uri given as radio uri and not as ip address. So if you change it here: https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/aideck/fpv.py#L71 to your ip address it should work (URI = 'tcp://192.168.4.1:5000' should be the default, but you might need to double check).
Be careful though - at least without gamepad connected this makes the motors turn on on full thrust, we did not try to use it properly - so let us know if it works for you or if you run in any more problems!
Hope this helps,
Hanna

**Reply by matze946** [reply]
Hi Hanna,
yes that worked for me thank you, I can now see the image stream from the crazyflie but I can't control it with the keyboard. The crazyflie also don't go on full thrust it just do nothing but streaming the images. Any idea how I can cotrol it now?
Matthias

**Reply by hmllr** [reply]
Actually, forget the thing with full thrust, I think it just seemed that way in the test we did (which is completely unrelated). I cannot test it right now, but I think it should actually just work. I guess it works only with a flow deck mounted though, otherwise holding the position is too difficult. And I guess you don't have a flow deck mounted and that is why it is not taking off - but I might be wrong ;)
Hope this helps,
Hanna

**Reply by matze946** [reply]
I have the flow deck mounted to the crazyflie. Maybe my output helps

`INFO:cflib.crazyflie:Callback->Connection initialized[tcp://192.168.4.1:5000]
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
CPX socket transport
Connecting to socket on 192.168.4.1:5000...
Connected
Creating queue for CPXFunction.CRTP
INFO:cflib.crazyflie:We are connected[tcp://192.168.4.1:5000], request connection setup
INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
LINK:  <cflib.crtp.tcpdriver.TcpDriver object at 0x7f7b3cb43520>
Creating queue for CPXFunction.APP
INFO:cflib.crazyflie:Closing link
Closing transport
Driver closed
INFO:cflib.crazyflie:Callback->Disconnected from [tcp://192.168.4.1:5000]
Disconnected`

Could it be because I don't have the connection with the radio signal anymore?

**Reply by hmllr** [reply]
Hmm, I think the packets should just be sent over wifi, that shouldnt be a problem. 
Can you fly with your Crazyflie normally? Like if you connect with the cfclient over radio and press take off?

**Reply by matze946** [reply]
Yes I can fly via the take off button.

**Reply by hmllr** [reply]
I just tested it, and for me it works fine.
I am using the latest release of everything (the gap8 firmware is not in the firmware release, but there I am also using the latest) - can you update everything (also the python lib) and try again?

**Reply by mjm41** [reply]
Just curious, why does fpv.py line 71 have the default radio address even though we have to change it to our IP for it to work. Why not update the github so line 71 just says: `URI = 'tcp://192.168.4.1:5000'`?

**Reply by knmcguire** [reply]
@mjm41 

Could you make a new discussion for this? that would be easier for us to answer and more clearer for the rest of the users to see the support

**Reply by hmllr** [reply]
or just directly do a pull request and become a contributor :)  I must have missed that when I answered this discussion, if you add an URI as an argument it overwrites the wrong default value. Or if you have an environmental variable set for it as well. Sorry @matze946 ! And good catch @mjm41 ! Replacing the default would be a good idea. However, keeping the option to overwrite/use the environmental variable is also important as you can have different IP addresses if you connect to an existing network.
